dns/doh.go

```diff
@@ -47,17 +38,17 @@ func GetDoHClient(upstream string) *DoHClient {
 				},
 			}
 
-			client = &DoHClient{
-				upstream: upstream,
-				c:        c,
+			client = &DOHClient{
+				upstream: "https://" + host + "/dns-query",
```

Comment:
This might be a bad idea. Since DoH providers do this (not a flippin' standard somehow, grrrrrrrr):
<img width="238" alt="image" src="https://github.com/user-attachments/assets/faf3862d-1714-4d44-9728-063479991379">
<img width="498" alt="image" src="https://github.com/user-attachments/assets/a2955720-4771-45cf-af48-8b52c9048e6e">

Making the user guess that they need to strip the URL might not be intuitive

Reply:
Yeah i agree with that. my bad.